src/get-started/first-java-service.md

```diff
@@ -0,0 +1,615 @@
+    classes:
+
+    1. Open context menu (right click on the directory)
+    2. "Mark Directory As" > "Generated Sources Root"
+
+Compile the project:
+
+```shell
+mvn compile
+```
+
+<!--
+todo: Shall we add a whole file of MySchema?
+-->
+
+<!--
+Todo: Shall we draw a parallel with the DAO objects in business
+applications?
+-->
+
+<!--
+Todo: Shall we include the references to extra info on MerkleDB,
+indexes, serialization and serializers?
+-->
+
+### 3 Service Transactions
+
+Our service needs two operations updating its state:
+
+- Add a new vehicle entry to the registry
+- Transfer the ownership over the vehicle to another user.
+
+Modifying operations in Exonum are called transactions. Transactions are
+implemented as methods in a service class — a class implementing
+[`Service`][service-interface] interface. A transaction method must be annotated
+with [`@Transaction`][transaction-annotation] annotation.
+
+[service-interface]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/service/Service.html
+[transaction-annotation]: https://exonum.com/doc/api/java-binding/0.10.0/com/exonum/binding/core/transaction/Transaction.html
+
+Our project already has a template service `MyService` — navigate to it.
+
+#### Add Vehicle Transaction
+
+First, let's define the transaction arguments. As Exonum expects transaction
+arguments in a serialized form, we will define the arguments as a protobuf
+message.
+
+Add a new file `transactions.proto` in `car-registry-messages`
+(`car-registry-messages/src/main/proto/example/vehicle`).
+
+<!--
+todo: Again, shall we include a code example of an empty file
+(with package, option java_package, etc.)?
+-->
+
+Then add the message definition of the _Add Vehicle_ transaction:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-messages/src/main/proto/example/vehicle/transactions.proto)
+inside_block:ci-add-vehicle
+<!--/codeinclude-->
+
+Compile the message:
+
+```shell
+mvn generate-sources
+```
+
+---
+
+Next, let's write the transaction method. Navigate back to `MyService`,
+and add the following method:
+
+<!--codeinclude-->
+[](../../code-examples/java/exonum-java-binding/tutorials/car-registry/car-registry-service/src/main/java/com/example/car/MyService.java)
+inside_block:ci-add-vehicle
+<!--/codeinclude-->
+
+The annotation accepts an integral _transaction ID_: an identifier of
+the transaction that service clients use to invoke it.
+
+The method accepts `AddVehicle` class: an auto-generated class from our
+protobuf definition. It also accepts the `TransactionContext` allowing
+the transaction to access the database.
+
+The transaction implementation uses the service data schema `MySchema`
+to access its data.
+
+Note that the transaction throws an `ExecutionException` in case of
+a precondition failure: in this case, an attempt to add a vehicle with
+an existing ID.
```

Comment:
It could be worth explaining what throwing `ExecutionException` accomplishes (rolls back all changes made by the transaction and records its status as erroneous).

Reply:
Added and also linked the section in the guide.